File: C/牛客网/剑指offer/item062.cpp
#include <iostream>

using namespace std;

struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
		val(x), left(NULL), right(NULL) {
	}
};


class Solution {
public:
	TreeNode* KthNode(TreeNode* pRoot, int k) {
		if (pRoot) {
			TreeNode* node;
			node = KthNode(pRoot->left, k);
			if (node) return node;
			if (++count == k) return pRoot;
			node = KthNode(pRoot->right, k);
			if (node) return node;
		}
		return NULL;
	}

	int count = 0;
};

File: Java/LeetCode/100/075/item062.java
class Solution {
    public int uniquePaths(int m, int n) {
        if (m > n) return uniquePaths(n, m);
        long count = 1;
        for (int i = n; i < m + n - 1; i++) count *= i;
        for (int i = 2; i < m; i++) count /= i;
        return (int) count;
    }
}